src/System.Text.RegularExpressions/tests/RegexCultureTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexCultureTests
+    {
+        /// <summary>
+        /// See https://en.wikipedia.org/wiki/Dotted_and_dotless_I 
+        /// </summary>
+        [Fact]
+        public void TurkishI_Is_Differently_LowerUpperCased_In_Turkish_Culture()
+        {
+            var turkish = new CultureInfo("tr-TR");
+            string input = "Iıİi";
+
+            var cultInvariantRegex = Create(input, CultureInfo.InvariantCulture, RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+            var turkishRegex =       Create(input, turkish,                      RegexOptions.IgnoreCase);
```

Comment:
Nit: please change var to be the actual type here... we only use var when the type is obvious from the right-hand side, namely a ctor or an explicit cast.